### ISSUE TITLE
HTTPClient Docker Memory Leak

### ISSUE THREAD

**Issue by SentryMan** [reactor/reactor-netty#1304]
We're trying to use reactive in our microservice, but we're seeing the docker memory go up until the instance is killed. We were able to identify our API Calls as the culprit. 

## Expected Behavior

The memory usage should be similar or less than using the ``RestTemplate`` or the deprecated ``AsyncRestTemplate``.

## Actual Behavior

When we make API calls with anything that uses netty's ``HttpClient``, we're seeing the memory increase seemingly without limit. To be more precise, the JVM memory appears stable, and I was able to confirm the GC is happening with VisualVM. but when I look at the docker stats, the memory is consistently going up until  the instance is eventually killed without any outOfHeap errors.

## Steps to Reproduce

```java
@Component
public class Controller {

  private static final String ENDPOINT = "http://<your-local-ip>:9090";

   
  private HttpClient client;
  
  public Controller() {
   super();
   this.client = HttpClient.create();
   }
  
  @Bean
  public RouterFunction<ServerResponse> router() {

    return RouterFunctions.route(GET("helloworld"), this::handle);
  }

  Mono<ServerResponse> handle(ServerRequest request) {

    Flux<String> helloMono =
        client.get().uri(ENDPOINT + "/hello").responseContent().asByteArray().map(String::new);

    Flux<String> worldMono =
        client.get().uri(ENDPOINT + "/world").responseContent().asByteArray().map(String::new);

    return ServerResponse.ok().body(helloMono.zipWith(worldMono, (h, w) -> h + w), String.class);
  }
}

```

I've created sample projects for all the things I tried. Please check if you can reproduce the issue. 

Run the backend, then bring up your chosen repo in docker.(be sure to replace the endpoint with your local IP) Finally, run a Jmeter with the JMX inside the repo then open up the docker stats and see the memory usage.

The backend: [The Hello World Backend](https://github.com/SentryMan/api-hello-world-backend)

[The Webclient Hello World](https://github.com/SentryMan/reactive-hello-world)(JMX is inside this repo)
[The Exchange Strategy Hello World](https://github.com/SentryMan/reactive-hello-world/tree/ExchangeStrategy)
[The Netty HTTP Client Hello World](https://github.com/SentryMan/reactive-hello-world/tree/Netty-HTTP-Client)

//These repos behave as expected
[The RestTemplate Hello World](https://github.com/SentryMan/reactive-hello-world/tree/restTemplate)
[The AsyncRestTemplate Hello World](https://github.com/SentryMan/reactive-hello-world/tree/AsyncRestTemplate)

## Your Environment

* Reactor version(s) used: Spring Webflux 5.2.7; Reactor Core 3.3.6
* Reactor Netty version: 0.9.8
* OS and version (eg. `uname -a`): In prod we use our custom java 8 image. but for the example repos I use the openjdk:8 docker image.

**Comment by violetagg** [comment]
@SentryMan [Reactor Netty version: 0.9.8](https://github.com/reactor/reactor-netty/releases/tag/v0.9.8.RELEASE) has FDs leak. Can you try the latest available release (0.9.11)?

**Comment by violetagg** [comment]
@SentryMan `The Exchange Strategy Hello World` uses `exchange` and I would strongly recommend you to switch to `retrieve`.
https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html#webflux-client-exchange

```
Unlike retrieve(), when using exchange(), it is the responsibility of the application to consume any response content regardless of the scenario (success, error, unexpected data, etc). Not doing so can cause a memory leak.
```

https://github.com/spring-projects/spring-framework/issues/25751

**Comment by SentryMan** [comment]
> @SentryMan [Reactor Netty version: 0.9.8](https://github.com/reactor/reactor-netty/releases/tag/v0.9.8.RELEASE) has FDs leak. Can you try the latest available release (0.9.11)?

Thanks for responding, It appears that upgrading to 0.9.11 solved it.

**Comment by violetagg** [comment]
Duplicate for #1152

**Comment by SlandShow** [comment]
@violetagg can you pls verify reactor-netty **0.7.10.RELEASE** version? Looks like we faced the same issue...

**Comment by violetagg** [comment]
@SlandShow 0.7.10.RELEASE is not the latest released version from 0.7.x branch, so there is a chance this is fixed.
Version 0.7.x is not supported anymore.

**Comment by SlandShow** [comment]
@violetagg but anyway, we are faced with the next issue:
```ERROR [lettuce-epollEventLoop-6-2] io.netty.util.ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected```

So, our set-up:
1. lettuce `5.1.6.RELEASE` version
2. Spring boot `2.0.6.RELEASE`
3. Netty `4.1.29.Final`
4. Project reactor `io.projectreactor.ipc:reactor-netty:jar:0.7.10.RELEASE`

Mb you know what is the actual problem?
